core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -613,7 +613,7 @@ private void readCurrentBlockToPos(long pos) throws IOException {
     // Do not set the buffer size too small to avoid slowing down seek by too much.
     byte[] buffer = new byte[Math.min((int) mSeekBufferSizeBytes, (int) len)];
     do {
-      int bytesRead = read(buffer);
+      int bytesRead = read(buffer, 0, (int) Math.min(buffer.length, len));
```

Comment:
I believe that @peisun1115's fix addresses the following scenario:

1) `mSeekBufferSizeBytes` < `len` and so the `do` loop will execute multiple times
2) the last iteration of the loop will result in a read that reads past the end of the current block